src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -802,7 +802,7 @@ protected override void DecodeWellKnownAttributeImpl(ref DecodeWellKnownAttribut
             else if (ReportExplicitUseOfReservedAttributes(in arguments,
                 ReservedAttributes.DynamicAttribute |
                 ReservedAttributes.IsReadOnlyAttribute |
-                // PROTOTYPE: RequiresLocationAttribute
+                ReservedAttributes.RequiresLocationAttribute |
```

Comment:
>RequiresLocationAttribute [](http://example.com/codeflow?start=35&length=25)

I see ```ReservedAttributes.IsReadOnlyAttribute``` specified as reserved for other symbols too. Consider follow the pattern

Reply:
I think `IsReadOnly` has wider semantics - for example, returns can be readonly, but they probably cannot be `RequiresLocation`.

Reply:
> I think IsReadOnly has wider semantics - for example, returns can be readonly, but they probably cannot be RequiresLocation.

Looking at IsByRefLikeAttribute, compiler synthesizes it only for named types, but ReservedAttributes.IsByRefLikeAttribute is checked for a wide set of symbols. I don't think this is a blocking issue for this PR, but let's confirm with LDM whether we want proactively prohibit application of the attribute in other contexts in order to preserve future design space.

Reply:
Added to section "alternatives pending LDM review": https://github.com/dotnet/csharplang/pull/7328